### PR TITLE
Add resource group creation module - tag filter only

### PR DESCRIPTION
**Pull request by niroz89**

### CODE REVIEW

File: tests/setup/resources/versions.tf [hunk 1]
```diff
@@ -0,0 +1,10 @@
+terraform {
+  required_version = ">= 1.6"
+
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~> 5.38"
```
Comment (uchinda-sph) [inline tests/setup/resources/versions.tf:7]:
```suggestion
      version = ">= 5.38"
```

File: examples/simple/versions.tf [hunk 2]
```diff
@@ -0,0 +1,10 @@
+terraform {
+  required_version = ">= 1.6"
+
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~> 5.38"
```
Comment (uchinda-sph) [inline examples/simple/versions.tf:7]:
```suggestion
      version = ">= 5.38"
```

**Review by uchinda-sph — APPROVED** [review]

File: tests/setup/resources/versions.tf (hunk 1 above)
Comment (niroz89) [inline tests/setup/resources/versions.tf:7]: This is not the module's version requirement. This is setup code for tests. No need to be a more liberal operator.

File: examples/simple/versions.tf (hunk 2 above)
Comment (niroz89) [inline examples/simple/versions.tf:7]: This is not the module's version requirement. This is example code. No need to be a more liberal operator.

**Review by uchinda-sph — APPROVED** [review]

**Pull request merged**